src/metric.jl

```diff
@@ -98,6 +98,84 @@ function Base.show(io::IO, dem::DenseEuclideanMetric)
     return print(io, "DenseEuclideanMetric(diag=$(_string_M⁻¹(dem.M⁻¹)))")
 end
 
+"""
+    RankUpdateEuclideanMetric{T,AM,AB,AD,F} <: AbstractMetric
+
+A Gaussian Euclidean metric whose inverse is constructed by rank-updates.
+
+# Fields
+
+$(TYPEDFIELDS)
+
+# Constructors
+
+    RankUpdateEuclideanMetric(n::Int)
+    RankUpdateEuclideanMetric(M⁻¹, B, D)
+
+ - Construct a Gaussian Euclidean metric of size `(n, n)` with `M⁻¹` being diagonal matrix.
+ - Construct a Gaussian Euclidean metric of `M⁻¹`, where `M⁻¹` should be a full rank positive definite matrix,
+    and `B` `D` must be chose so that the Woodbury matrix `W = M⁻¹ + B D B^\\mathrm{T}` is positive definite.
```

Comment:
For the other metrics `M⁻¹` is defined as the inverse of the metric. Wouldn't it be more consistent to here define `M⁻¹ = A + B D B'` as the inverse metric and require A be diagonal and positive-definite? (also, "full-rank" for diagonal PD-mat is redundant).

Reply:
For Pathfinder to safely convert its `WoodburyPDMat` to this type, we'll need the contents/form of the `factorization` field to be documented.